### PR TITLE
bsp: add support for qemuarma15

### PR DESCRIPTION
**Pull request by mike-scott**
- import basic qemuarma15 support from yocto-kernel-cache
- adjust settings to support LmP

### CODE REVIEW

**Comment by mike-scott** [comment]
I should probably include the squashed patch descriptions from Yocto for the 1st commit:
https://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/log/bsp/qemuarma15?h=yocto-5.4

**Comment by ricardosalveti** [comment]
> I should probably include the squashed patch descriptions from Yocto for the 1st commit:
> https://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/log/bsp/qemuarma15?h=yocto-5.4

Yeah, and please also specify the hash of the latest commit used as reference, so we can easily go back in time if needed.

File: bsp/qemuarma15/qemuarma15-preempt-rt.scc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: MIT
```
Comment (ricardosalveti) [inline bsp/qemuarma15/qemuarma15-preempt-rt.scc:1]: We don't yet support rt, so fine to drop this file.

File: bsp/qemuarma15/qemuarma15-standard.scc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: MIT
+define KMACHINE qemuarma15
+define KTYPE standard
+define KARCH arm
+
+include ktypes/standard/standard.scc
+# we aren't carrying patches specific to this platform, so let's
+# skip the branch for now.
+# branch qemuarma15
```
Comment (ricardosalveti) [inline bsp/qemuarma15/qemuarma15-standard.scc:9]: Please remove these lines.

File: bsp/qemuarma15/qemuarma15-standard.scc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+include features/bluetooth/bluetooth.scc
+include features/bluetooth/bluetooth-usb.scc
+include features/bluetooth/bluetooth-uart.scc
+include features/ieee802154/ieee802154.scc
+include features/mac802154/mac802154.scc
+
+include cfg/virtio.scc
+include cfg/dmaengine.scc
+include cfg/timer/rtc.scc
```
Comment (ricardosalveti) [inline bsp/qemuarma15/qemuarma15-standard.scc:19]: Should we align this fine with our the other bsps we have? Most of the common include bits are generic.

File: bsp/qemuarma15/qemuarma15-tiny.scc [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: MIT
```
Comment (ricardosalveti) [inline bsp/qemuarma15/qemuarma15-tiny.scc:1]: We also don't yet support tiny, so feel free to drop this file as well.

File: bsp/qemuarma15/qemuarma15-preempt-rt.scc (hunk 1 above)
Comment (mike-scott) [reply bsp/qemuarma15/qemuarma15-preempt-rt.scc:1]: I wonder if dropping will affect future merges from yocto?

**Comment by mike-scott** [comment]
@ricardosalveti Squashed initial commit

Let me get the changes you requested in follow-up commits

**Comment by mike-scott** [comment]
@ricardosalveti Comments addressed.  Let me know if this needs any other changes.

**Pull request merged**